index.src.html

```diff
@@ -816,27 +831,29 @@ <h3 id=interface-reporting-observer>Interface {{ReportingObserver}}</h3>
   
   1. Set the <a>context object</a>'s <a>options</a> to |options|.
   
-  2. If it is not already present, add the <a>context object</a> to the 
-     list of <a>registered reporting observers</a> for the <a>environment
-     settings object</a> associated with the <a>context object</a>.
+  2. If it is not already registered, <a>register</a> the <a>context object</a>
+     with its associated <a>environment settings object</a>.
   
   The <dfn method for=ReportingObserver><code>disconnect()</code></dfn> method,
   when invoked, must run these steps:
-  
-  1. If present, remove the <a>context object</a> from the list of
-     <a>registered reporting observers</a> for the <a>environment settings
-     object</a> associated with the <a>context object</a>.
+
+  1. If the <a>context object</a> is not registered, stop.
+
+  2. Remove the <a>context object</a> from the <a>registered reporting observer
+     list</a> of its associated <a>environment settings object</a>.
+
+  3. Execute the <a href="#invoke-observers">invoke reporting observers</a>
```

Comment:
Why is this necessary?  In most cases a task is already queued when the report list is updated.

Is it just for the case where `futureOnly` was false and no other reports have been queued yet?  I don't think we want to delay callback invocation those reports until `disconnect`.  Instead `observe` should probably be queuing a task like this when the buffer starts with some entries.

Reply:
I think you're right, and this shouldn't be necessary. I will remove step 3 here and ensure a task is queued when buffered reports are added to new observers.